dartagnan/src/main/java/com/dat3m/dartagnan/expression/processing/ExprSimplifier.java

```diff
@@ -232,6 +275,26 @@ public Expression visitITEExpression(ITEExpr expr) {
             }
         }
 
+        if (trueCase instanceof BoolLiteral tLit && falseCase instanceof BoolLiteral fLit) {
+            if (tLit.getValue() == fLit.getValue()) {
+                if (aggressive || cond.getRegs().isEmpty()) {
+                    return tLit;
+                } else if (tLit.getValue()) {
+                    return expressions.makeBoolBinary(cond, BoolBinaryOp.OR, expressions.makeTrue());
+                } else {
+                    return expressions.makeBoolBinary(cond, BoolBinaryOp.AND, expressions.makeFalse());
```

Comment:
Yes, but I won't put an extra comment there because the whole class preserves dependencies (if `aggressive=false`) and so I would have to put that comment into every single optimization if we want to be consistent.

Reply:
I was worried someone would read the `\/ true` or `/\ false` and get completely lost (as I did when I just saw the diff), but the class is documented at the top about this, so this should be enough